### ISSUE TITLE
Use Case: Accessibility Notifications

### ISSUE THREAD

**Issue by cookiecrook** [a11y-api/a11y-api#3]
**Use Case: Accessibility Notifications**

Currently, web authors primarily alert assistive technologies to important document change events by moving focus around. For example, new content in a single-page web application is loaded so the author forces focus there. This means ATs have to heuristically determine what happened and why it's relevant to the user. iOS has several author-supplied notifications that would be useful in the context of web accessibility.

Announcement (e.g. Speak this text) and the callback AnnouncementDidFinish
LayoutChanged
ScreenChanged (single-page web apps could trigger this when loading new content)

**Comment by minorninth** [comment]
My worry about Announcement is that it doesn't necessarily map as well to things like Braille or other types of AT. It's tempting for authors to just make a self-voicing app that works okay with a speech-based screen reader but doesn't work with any other AT.

I'm not saying we should disallow it, but it'd be a good idea to make sure authors use it sparingly. It'd be nice if we could fire it on an element whenever possible, kind of like a live region or alert, but basically saying that this element should announce itself now.

LayoutChanged - needed for a virtual view hierarchy, shouldn't be needed for existing HTML DOM

ScreenChanged - yes, this would be good. Maybe a special case of Announcement?
